### PR TITLE
add --name to appsody stop 

### PR DESCRIPTION
**Pull request by kewegner**
Added --name parameter to appsody stop

### CODE REVIEW

**Comment by kewegner** [comment]
#37 is resolved with this PR.  A test case is added for appsody stop as well.

File: cmd/stop_test.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+		runChannel <- err
+	}()
+
+	// defer the appsody stop to close the docker container
+	defer func() {
+
+		stopOutput, errStop := cmdtest.RunAppsodyCmdExec([]string{"stop"}, projectDir)
+
+		//docker[stop appsody-stop-test
+		if !strings.Contains(stopOutput, "docker[stop appsody-stop-test") {
```
Comment (kylegc) [inline cmd/stop_test.go]: Checking that the stop command ran is okay, but I think we really should run a `docker ps` to verify the container was stopped.

File: cmd/stop.go [hunk 2]
```diff
@@ -32,14 +32,19 @@ var stopCmd = &cobra.Command{
...
+		}
 		//dockerRemove(imageName) is not needed due to --rm flag
 		os.Exit(1)
 	},
 }
+var stopContainerName string
 
 func init() {
 	rootCmd.AddCommand(stopCmd)
+	stopCmd.PersistentFlags().StringVar(&stopContainerName, "name", "", "The name of your development container.")
```
Comment (kylegc) [inline cmd/stop.go]: We should try to reuse the `name` flag from dev_common. See the addDevCommonFlags() func as an example. We'll need to pull out the name from there into a different common flag.

File: cmd/cmdtest/testutils.go [hunk 3]
```diff
@@ -209,3 +209,67 @@ func AddLocalFileRepo(repoName string, repoFilePath string) (string, func(), err
 
 	return repoURL, cleanupFunc, err
 }
+
+// RunDockerCmdExec runs the docker command with the given args in a new process
+// The stdout and stderr are captured, printed, and returned
+// args will be passed to the docker command
+// workingDir will be the directory the command runs in
+func RunDockerCmdExec(args []string, workingDir string) (string, error) {
+	execDir, err := os.Getwd()
```
Comment (kylegc) [inline cmd/cmdtest/testutils.go]: `docker` commands don't rely on the working directory, so you can remove all the code related to this.

File: cmd/stop.go [hunk 4]
```diff
@@ -32,14 +32,18 @@ var stopCmd = &cobra.Command{
 		Info.log("Stopping development environment")
 		projectDir := getProjectDir()
 		imageName := fmt.Sprintf("%s-dev", filepath.Base(projectDir))
-
-		dockerStop(imageName)
+		if containerName == "" {
+			dockerStop(imageName)
```
Comment (kylegc) [inline cmd/stop.go]: containerName will always be set, you can remove this if check and the imageName and always use containerName

**Pull request merged**